lang/src/org/partiql/lang/CompilerPipeline.kt

```diff
@@ -198,10 +199,15 @@ private class CompilerPipelineImpl(
     override fun compile(query: ExprNode): Expression {
         val context = StepContext(valueFactory, compileOptions, functions, procedures)
 
-        val preProcessedQuery = preProcessingSteps.fold(query) { currentExprNode, step ->
-            step(currentExprNode, context)
-        }
+        val preProcessedQuery = executeProcessingSteps(query, context)
 
         return compiler.compile(preProcessedQuery)
     }
+
+    internal fun executeProcessingSteps(query: ExprNode, context: StepContext): ExprNode {
+        val preProcessedQuery = preProcessingSteps.interruptibleFold(query) { currentExprNode, step ->
```

Comment:
nit: could be inlined.

```
return preProcessingSteps.interruptibleFold(query) { currentExprNode, step ->
    step(currentExprNode, context)
}
```

Reply:
actually, the entire function should be inlined at line #202.